docs/ui-style-guide.md

```diff
@@ -0,0 +1,67 @@
+# Machinery UI Style Guide
+
+Machinery has two user facing interfaces, the command line and the graphical user interface in the form of an HTML view.  
+All additions to Machinery should follow the same or at least a similar style to allow a consistent user experience.  
+This guide should provide some guidelines and examples for the UI.
+
+## Command Line Interface
+
+### Output
+
+The Markdown syntax is used to format the data when applicable, for example for headlines, listings and so on:
+
+```
+# Services [example] (2016-07-05 17:07:38)
+
+  * after-local.service: static
+  * auditd.service: enabled
+  * auth-rpcgss-module.service: static
+  * autovt@.service: disabled
+
+```
+
+Additional attributes of a list are bracketed, if it would still fit one line of ~80 characters:
+```
+* bin (bin, uid: 1, gid: 1, shell: /bin/bash)
+```
+```
+*util-linux-2.25-12.2.x86_64 (openSUSE)
+```
+
+If not, the data is formatted as a list entry where the following lines are indented on the same level as the text of the first line:
+```
+  * /boot/backup_mbr (file)
+    User/Group: root:root
+    Mode: 644
+    Size: 512 B
+
+```
+
+### Command Names
+
+Commands should be only named with one word if possible. If more than one word is needed it must be separated with a dash.
+
+### Documentation
+
+Whole sentences with dots at the end should be used for all documentation and help texts.
+
+The only exceptions are descriptions in one line of a simple parameter in the command line help.
+
+### Errors
+
+Possible user errors should be captured and handled. If not the default is shown which would provide a backtrace and a message for the user to report the issue.
+
+For error messages the method `Machinery::Ui.error("Error: <message>")` needs to be used and it should be a regular sentence with a dot at the end.
```

Comment:
I might even consider using active voice here, like `For error messages use the method..., which should contain a regular...` unless it doesn't seem consistent with the rest of the guide.